server/app/services/rpc/container_handler.rb

```diff
@@ -115,18 +109,36 @@ def event(data)
           container.destroy
         end
       end
+
+      {}
+    end
+
+    def flush_logs
+      @db_session[:container_logs].insert(@logs)
+      @logs.clear
+    end
+
+    def flush_stats
+      @db_session[:container_stats].insert(@stats.dup)
+      @stats.clear
+    end
+
+    def gc_cache
+      if @cached_containers.keys.size > @containers_cache_size
+        (@containers_cache_size / 5).times { @cached_containers.shift }
```

Comment:
It seems to me as if `Hash#shift` is deterministic: it will always remove the oldest value from the hash. That would mean that this will behave as a kind of ordered queue, which is okay, I guess, in that it ensures that it can't get clogged with dead containers.

Ideally this would use some more effective cache... some basic LRU would be more effective at keeping a small number of very heavy-logging containers in the cache.... this cache will quickly flush out any heavy loggers if you also have lots of quieter ones?